app/models/ZkKafka.scala

```diff
@@ -40,20 +40,26 @@ object ZkKafka {
   def getTopologies: Seq[Topology] = {
     zkClient.getChildren.forPath(makePath(Seq(stormZkRoot))).asScala.map({ r =>
       getSpoutTopology(r)
-    }).sortWith(topoCompFn)
+    }).flatten.sortWith(topoCompFn)
   }
 
-  def getSpoutTopology(root: String): Topology = {
+  def getZkData(path: String): Option[Array[Byte]] = Option(zkClient.getData.forPath(path))
+
+  def getSpoutTopology(root: String): Option[Topology] = {
     // Fetch the spout root
     val s = zkClient.getChildren.forPath(makePath(applyBase(Seq(stormZkRoot, Some(root)))))
     // Fetch the partitions so we can pick the first one
     val parts = zkClient.getChildren.forPath(makePath(applyBase(Seq(stormZkRoot, Some(root))) ++ Seq(Some(s.get(0)))))
     // Use the first partition's data to build up info about the topology
-    val jsonState = new String(zkClient.getData.forPath(makePath(applyBase(Seq(stormZkRoot, Some(root))) ++ Seq(Some(s.get(0))) ++ Seq(Some(parts.get(0))))))
-    val state = Json.parse(jsonState)
-    val topic = (state \ "topic").as[String]
-    val name = (state \ "topology" \ "name").as[String]
-    Topology(name = name, topic = topic, spoutRoot = root)
+    // Also, don't trust zk to have valid JSON, it may be null or something...
+    val path = makePath(applyBase(Seq(stormZkRoot, Some(root))) ++ Seq(Some(s.get(0))) ++ Seq(Some(parts.get(0))))
```

Comment:
I didn't change this, aside from splitting `path` out so we could pass that into `getZkData`... and I honestly don't know why you would add a `.map` there. A hangout is probably in order to explain that :)